### PR TITLE
Add Gateway API support doc

### LINKED ISSUE #500: docs: Describe which areas of Gateway API are and are not supported
This information is buried in the conformance-test output, but it should be made much more user-visible.

### PR DESCRIPTION
**Pull request by AliceProxy**
Signed-off-by: AliceProxy <alicewasko@datawire.io>

Adds a simple doc which goes over which Gateway API resources Envoy Gateway supports with a short description about each containing links to the Gateway API docs.

Fixes #500

### CODE REVIEW

File: docs/v0.2.0/design/gatewayapi-support.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# Gateway API Support
+
+As mentioned in the [SystemDesign][] document, Envoy Gateway's managed data plane is configured dynamically through 
+Kubernetes resources, primarily [Gateway API][] objects. Envoy Gateway supports configuration using the following Gateway API resources.
+
+## **GatewayClass**
+
+[GatewayClasses][] are used to configure which Gateways and other reliant resources should be managed by Envoy Gateway.
```
Comment (arkodg) [inline docs/v0.2.0/design/gatewayapi-support.md:8]: can we add a comment mentioning that EG supports/accepts a single GatewayClass resource linked to the EG controller, and accepts in order of age (oldest first)

File: docs/v0.2.0/design/gatewayapi-support.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+- `responseHeaderModifier`: [ResponseHeaderModifiers](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter) can be used to modify or add response headers before the response is sent back to the client.
+- `requestMirror`: [RequestMirrors](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter) configure destinations where the requests should also be mirrored to. Responses to mirrored requests will be ignored.
+- `requestRedirect`: [RequestRedirects](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter) configure policied for how requests that match the HTTPRoute should be modified and then redirected.
+- `urlRewrite`: [UrlRewrites](https://gateway-api.sigs.k8s.io/references/spec/#gateway.networking.k8s.io/v1beta1.HTTPRouteFilter) allow for modification of the request's hostname and path before it is proxied to its destination.
+- filters...
+
```
Comment (arkodg) [inline docs/v0.2.0/design/gatewayapi-support.md:27]: would be good to add `extensionRef` in here and link it to Auth and RL

**Comment by arkodg** [comment]
thanks for adding this @AliceProxy, shouldnt the docs be added under `docs/design/latest` instead ?

**Comment by danehans** [comment]
CI is failing due to https://github.com/envoyproxy/gateway/issues/1016 flake.

**Comment by danehans** [comment]
> thanks for adding this @AliceProxy, shouldnt the docs be added under docs/design/latest instead ?

Agreed, please move the doc to `docs/latest/design/gatewayapi-support.md`.

File: docs/v0.2.0/design/gatewayapi-support.md (hunk 1 above)
Comment (danehans) [reply docs/v0.2.0/design/gatewayapi-support.md:8]: s/[GatewayClasses][] are/A [GatewayClass][] is/

File: docs/v0.2.0/design/gatewayapi-support.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+**Note:** currently the only kind of [BackendRef][] kind (the destination where traffic should be sent to) that Envoy Gateway supports are Kubernetes services. Routing traffic to other destinations such as arbitrary URLs is not currently possible.
```
Comment (danehans) [inline docs/v0.2.0/design/gatewayapi-support.md:28]: ~kind of~

File: docs/v0.2.0/design/gatewayapi-support.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+As mentioned in the [SystemDesign][] document, Envoy Gateway's managed data plane is configured dynamically through 
+Kubernetes resources, primarily [Gateway API][] objects. Envoy Gateway supports configuration using the following Gateway API resources.
+
+## **GatewayClass**
+
+[GatewayClasses][] are used to configure which Gateways and other reliant resources should be managed by Envoy Gateway.
+
+## **Gateway**
+
+When a [Gateway][] resource is created that references the GatwewayClass Envoy Gateway is managing then Envoy Gateway will
```
[the remark this answers is not in the document]
Comment (danehans) [reply docs/v0.2.0/design/gatewayapi-support.md:12]: Please add the gateway notes from https://github.com/envoyproxy/gateway/issues/500#issuecomment-1421563543 here.

**Comment by danehans** [comment]
Waiting for https://github.com/envoyproxy/gateway/pull/1021 to merge and will then rerun CI.

**Review by arkodg — APPROVED** [review]
LGTM thanks @AliceProxy ! this doc adds a ton of clarity for the end user

**Review by danehans — APPROVED** [review]
I still see a few nits, but I'll resolve them in a follow-up PR.

**Pull request merged**